core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -655,15 +655,19 @@ private[spark] object Utils extends Logging {
       // created the directories already, and that they are secured so that only the
       // user has access to them.
       getYarnLocalDirs(conf).split(",")
+    } else if (conf.getenv("SPARK_LOCAL_DIRS") != null) {
```

Comment:
@dragos Yes, makes sense. @Zariel can you move the `SPARK_LOCAL_DIRS` below `SPARK_EXECUTOR_DIRS` as others have suggested?